### PR TITLE
(Android only) Force audio through speaker

### PR DESCRIPTION
**Pull request by arturzaitcev-tomtom**
- Update build.gradle to be able to open android project in Android Studio
- Add a function to force audio through speaker
- Remove unused code

### CODE REVIEW

**Comment by arturzaitcev-tomtom** [comment]
Will make a tag when PR is ok :)

File: android/build.gradle [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+  }
+}
+
+apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion rootProject.hasProperty('compileSdkVersion') ? rootProject.compileSdkVersion : DEFAULT_COMPILE_SDK_VERSION
-    buildToolsVersion rootProject.hasProperty('buildToolsVersion') ? rootProject.buildToolsVersion : DEFAULT_BUILD_TOOLS_VERSION
+  compileSdkVersion safeExtGet('compileSdkVersion', 31)
+  buildToolsVersion '30.0.3'
```
Comment (casparvanderheiden-tomtom) [inline android/build.gradle]: Shouldn't you do the `safeExtGet` check here as well?

File: android/build.gradle [hunk 2]
```diff
@@ -19,6 +30,10 @@ android {
...
 
+  repositories {
+    mavenCentral()
+  }
+
 dependencies {
-  compileOnly 'com.facebook.react:react-native:${project.reactNativeVersion}'
-}
+  implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
+}
```
Comment (casparvanderheiden-tomtom) [inline android/build.gradle]: no newline at end of file

**Review by casparvanderheiden-tomtom — APPROVED** [review]
Approved with minor comments

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 3]
```diff
@@ -576,19 +578,9 @@ public void getCurrentTime(final Double key, final Callback callback) {
...
-    if (player != null) {
-      AudioManager audioManager = (AudioManager)this.context.getSystemService(this.context.AUDIO_SERVICE);
-      if(speaker){
-        audioManager.setMode(AudioManager.MODE_IN_COMMUNICATION);
-      }else{
-        audioManager.setMode(AudioManager.MODE_NORMAL);
-      }
-      audioManager.setSpeakerphoneOn(speaker);
-    }
+  public void setSpeakerphoneOn(final Boolean forcePhoneSpeaker) {
```
Comment (johanvanderboom-tomtom) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: Could you perhaps rename this from `setSpeakerPhoneOn` to something like `forcePhoneSpeaker`?

SpeakerPhone is term related to audio routing during a phone call (and what the original implementation was setting up) and has a different meaning than what we now use it to even though both are kind of trying to achieve a similar result moving to play over the speaker while another output is active the context is a bit different as we are not doing this during a call.

**Review by johanvanderboom-tomtom — APPROVED** [review]
approved with minor optional name change request

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java (hunk 3 above)
Comment (arturzaitcev-tomtom) [reply android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: Yes I can ;)

**Review by johanvanderboom-tomtom — APPROVED** [review]

**Pull request merged**